apps/src/aichat/views/modelCustomization/PublishNotes.tsx

```diff
@@ -24,24 +29,34 @@ const PublishNotes: React.FunctionComponent = () => {
     state => state.aichat.currentAiCustomizations
   );
 
-  const onUpdate = useCallback(
-    () => dispatch(updateAiCustomization()),
-    [dispatch]
-  );
+  const onSave = useCallback(() => {
+    dispatch(updateAiCustomization());
+    if (!hasFilledOutModelCard(modelCardInfo)) {
+      dispatch(setHasPublished(false));
+    }
+  }, [dispatch, modelCardInfo]);
 
-  const getInputTag = (property: keyof ModelCardInfo) => {
-    return property === 'botName' ? 'input' : 'textarea';
-  };
+  const onPublish = useCallback(() => {
+    dispatch(updateAiCustomization());
+    dispatch(setHasPublished(true));
+    dispatch(setViewMode(ViewMode.PRESENTATION));
```

Comment:
When the user clicks on 'Publish' and then the 'User View' is displayed, I imagine that the user may want to immediately update Model Card fields and click on 'Edit Mode'. What do you think about saving state so that the 'Publish' tab and not 'Setup' remains open to encourage an iterative process? Also related to comment below, that 'User View' remains displayed?

Reply:
That would be great as a follow-up and +1 on using Denys's new Tabs component - thanks!

Reply:
>Also related to comment below, that 'User View' remains displayed?
>Not sure I'm following this part, happy to chat offline if easier!

Ah sorry. I think once we save `hasPublished` to sources, this will not be an issue when a user moves from level to level. I think I was referring to how when a model card has been published, the toggle remain visible.